tests/src/test/scala/whisk/core/containerpool/logging/SplunkLogStoreTests.scala

```diff
@@ -0,0 +1,207 @@
+    annotations = Parameters("limits", ActionLimits(TimeLimit(1.second), MemoryLimit(128.MB), LogLimit(1.MB)).toJson),
+    duration = Some(123))
+
+  implicit val ec = system.dispatcher
+  implicit val materializer = ActorMaterializer()
+
+  val splunkIndex = system.settings.config.getString("whisk.logstore.splunk.index")
+  val activationIdFieldName = system.settings.config.getString("whisk.logstore.splunk.activation-id-field")
+  val logMessageFieldName = system.settings.config.getString("whisk.logstore.splunk.log-message-field")
+  val splunkUser = system.settings.config.getString("whisk.logstore.splunk.user")
+  val splunkPwd = system.settings.config.getString("whisk.logstore.splunk.password")
+  val splunkHost = system.settings.config.getString("whisk.logstore.splunk.host")
+  val splunkPort = system.settings.config.getInt("whisk.logstore.splunk.port")
+
+  val testFlow: Flow[(HttpRequest, Promise[HttpResponse]), (Try[HttpResponse], Promise[HttpResponse]), NotUsed] =
+    Flow[(HttpRequest, Promise[HttpResponse])]
+      .mapAsyncUnordered(1) {
+        case (request, userContext) =>
+          //handle assertion errors so that stream does not swallow error info
+          try {
+            request.uri.path.toString() shouldBe "/services/search/jobs"
+            request.headers shouldBe List(Authorization.basic(splunkUser, splunkPwd))
+            //we use cachedHostConnectionPoolHttps so won't get the host+port with the request
+            Unmarshal(request.entity).to[FormData].map { form =>
+              //handle assertion errors so that stream does not swallow error info
+              try {
+                val earliestTime = form.fields.get("earliest_time")
+                val latestTime = form.fields.get("latest_time")
+                val outputMode = form.fields.get("output_mode")
+                val search = form.fields.get("search")
+                val execMode = form.fields.get("exec_mode")
+
+                earliestTime shouldBe Some(startTime)
+                latestTime shouldBe Some(endTime)
+                outputMode shouldBe Some("json")
+                execMode shouldBe Some("oneshot")
+                search shouldBe Some(
+                  s"""search index="${splunkIndex}"| spath ${activationIdFieldName} | search ${activationIdFieldName}=${activation.activationId.toString} | table ${logMessageFieldName}""")
+
+                (
+                  Success(
+                    HttpResponse(
+                      StatusCodes.OK,
+                      entity = HttpEntity(
+                        ContentTypes.`application/json`,
+                        """{"preview":false,"init_offset":0,"messages":[],"fields":[{"name":"log_message"}],"results":[{"log_message":"some log message"},{"log_message":"some other log message"}], "highlighted":{}}"""))),
+                  userContext)
+              } catch {
+                case e: Throwable =>
+                  (Failure(e), userContext)
+              }
```

Comment:
Do you need the second try/catch? Seems like this would also translate to:

```scala
Unmarshal(request.entity).to[FormData].map { form =>
  // asserts
  (Success(HttpResponse(...)), userContext)
}.recover { case e =>
  (Failure(e), userContext)
}
```

Reply:
yes, I was thrown off by my IDE incorrectly hinting a compile error at `(Failure(e), userContext)`. Ignoring that does wonders to simplify this (and it works fine too).